### DISCUSSION TITLE
"gdk_seat_get_keyboard: assertion 'GDK_IS_SEAT (seat)' failed" if genie uninstalled

### DISCUSSION THREAD

**Discussion by ekrogh** [arkane-systems/genie#308] [Q&A]
Hello

If I uninstall genie and enable the native systemd 
```
(In /etc/wsl.conf:
[boot]
systemd=true
)

```

Then suddenly I get this error when starting gnome-terminal:
`gdk_seat_get_keyboard: assertion 'GDK_IS_SEAT (seat)' failed
`

If I don't uninstall genie before enabling the native systemd, then I don't get that error.

Wonder why that is ?
--
Eigil

**Comment by cerebrate** [comment]
That's because you haven't got a systemd user session set up via _sd-pam_ and _systemd-logind_. GDK (and other things that use the notion of seats) depend on that to get the information they need to operate correctly. Genie, by creating login sessions via the _systemd-logind_ mechanisms, ensures that all that (the seat, the systemd user session, the session dbus, etc., etc.) gets created. The native WSL _systemd_ support, in its current iteration, doesn't.

tl;dr this is what I created [bottle-imp](https://github.com/arkane-systems/bottle-imp) to address.

**Comment by ekrogh** [comment]
Now it looks like it is only if I upgrade ubuntu from ver. 22.04 to 22.10 that I have the problem.

Sorry I'm so imprecise but sometimes it works sometimes it doesn't.
